src/include/network/routines.rb

```diff
@@ -300,88 +352,42 @@ def UsedNicName(name)
+          log.error "Unknown network controller type: #{hwdevice}"
+          log.error "It's probably missing in hwinfo (NOT src/hd/hd.h:sc_net_if)" if subclass_id == 128
+        end
+        ret || ""
+      end
+    end
+
     # Simple convertor from subclass to controller type.
     # @param [Hash] hwdevice map with card info containing "subclass"
     # @return short device name
     # @example ControllerType(<ethernet controller map>) -> "eth"
     def ControllerType(hwdevice)
-      hwdevice = deep_copy(hwdevice)
-      return "modem" if Ops.get_string(hwdevice, "subclass", "") == "Modem"
-      return "isdn" if Ops.get_string(hwdevice, "subclass", "") == "ISDN"
-      return "dsl" if Ops.get_string(hwdevice, "subclass", "") == "DSL"
+      class_id    = hwdevice["class_id"]
+      subclass_id = hwdevice["sub_class_id"]
 
-      subclass_id = Ops.get_integer(hwdevice, "sub_class_id", -1)
+      return "modem" if hwdevice["subclass"] == "Modem" || modem_controller?(class_id)
+      return "isdn"  if hwdevice["subclass"] == "ISDN" || isdn_controller?(class_id)
+      return "dsl"   if hwdevice["subclass"] == "DSL" || dsl_controller?(class_id)
 
       # Network controller
-      if Ops.get_integer(hwdevice, "class_id", -1) == 2
-        if subclass_id == 0
-          return "eth"
-        elsif subclass_id == 1
-          return "tr"
-        elsif subclass_id == 2
-          return "fddi"
-        elsif subclass_id == 3
-          return "atm"
-        elsif subclass_id == 4
-          return "isdn"
-        elsif subclass_id == 6
-          return "ib"
-        elsif subclass_id == 129
-          return "myri"
-        elsif subclass_id == 130
-          return "wlan"
-        elsif subclass_id == 131
-          return "xp"
-        elsif subclass_id == 134
-          return "qeth"
-        elsif subclass_id == 135
-          return "hsi"
-        elsif subclass_id == 136
-          return "ctc"
-        elsif subclass_id == 137
-          return "lcs"
-        elsif subclass_id == 142
-          return "ficon"
-        elsif subclass_id == 143
-          return "escon"
-        elsif subclass_id == 144
-          return "iucv"
-        elsif subclass_id == 145
-          return "usb" # #22739
-        elsif subclass_id == 128
-          # Nothing was found
-          Builtins.y2error("Unknown network controller type: %1", hwdevice)
-          Builtins.y2error(
-            "It's probably missing in hwinfo (NOT src/hd/hd.h:sc_net_if)"
-          )
-          return ""
-        else
-          # Nothing was found
-          Builtins.y2error("Unknown network controller type: %1", hwdevice)
-          return ""
-        end
-      end
+      return network_subclass(hwdevice) if network_controller?(class_id)
+
       # exception for infiniband device
-      if Ops.get_integer(hwdevice, "class_id", -1) == 12
-        return "ib" if subclass_id == 6
-      end
+      return "ib" if infiniband_controller?(class_id) && subclass_id == 6
```

Comment:
why is not this integrated into ```network_controller?``` method? Infiniband is kind of network controller ....